src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java

```diff
@@ -141,8 +139,34 @@ public void setServerName(String serverName) {
      *
      * @return true if valid; false otherwise
      */
-    public boolean validate() {
-        return checkBaseUrl(baseUrl).kind != Kind.ERROR && checkServerName(serverName).kind != Kind.ERROR;
+    public FormValidation validate() {
+        return FormValidation.aggregate(Arrays.asList(checkBaseUrl(baseUrl), checkServerName(serverName), checkAdminCredentialsId(adminCredentialsId)));
+    }
+
+    /**
+     * Validates the provided admin credentials are present and appropriate
+     *
+     * @param adminCredentialsId
```

Comment:
Generally we don't need javadoc on private methods. Any reason you've chosen to do so here?
If we are having javadoc here, then can the description please be filled out for the param and return?

Reply:
Will do. I added it to be consistent with the other validation methods in this class.